test/nodetool/test_status.py

```diff
@@ -510,3 +510,40 @@ def test_status_resolve(request, nodetool, resolve):
     ]
 
     _do_test_status(request, nodetool, None, nodes, resolve)
+
+
+def test_status_with_zero_token_nodes(request, nodetool):
+    nodes = [
+        Node(
+            endpoint="127.0.0.1",
+            host_id="78a9c1d0-b341-467e-a076-9eff4cf7ffc6",
+            load=206015,
+            tokens=["-9175818098208185248", "-3983536194780899528"],
+            datacenter="datacenter1",
+            rack="rack1",
+            status=NodeStatus.Up,
+            state=NodeState.Normal,
+        ),
+        Node(
+            endpoint="127.0.0.2",
+            host_id="ed341f60-b12a-4fd4-9917-e80977ded0f9",
+            load=277624,
+            tokens=[],
+            datacenter="datacenter1",
+            rack="rack2",
+            status=NodeStatus.Up,
+            state=NodeState.Joining,
```

Comment:
`Moving` state was removed a long time ago from Cassandra and I don't know if it was ever implemented in Scylla. Ignore it.

Reply:
As discussed, we are not going forward with joining node display info. For leaving nodes, case is different, its not computed using tokens, it uses a getter for _leaving_endpoints data member, which is populated in `sync_raft_topology_nodes` . Hence it shows info regarding both regular nodes and zero token nodes

Reply:
Right, leaving zero-token nodes are fine.

Now I remember that we call `token_metadata::add_leaving_endpoint` for zero-token nodes. The same goes for `token_metadata::add_replacing_endpoint`. It looks bad because logically zero-token nodes have nothing to do with the token ring. I decided to leave it like this when I worked on zero-token nodes because it's harmless and changing it would require more refactoring. The zero-token nodes PR was big enough without it.

I don't think we want to clean up this in the near future, but maybe we should at least open an issue to document it. @kbr-scylla